lib/reply.js

```diff
@@ -427,9 +473,20 @@ function onSendEnd (reply, payload) {
   }
 
   res.writeHead(statusCode, reply[kReplyHeaders])
+  // write payload first
+  res.write(payload)
+
+  if (reply[kReplyShouldSendTrailers]) {
+    // compute the trailers
+    const trailers = {}
+    for (const trailerName of trailerHeaders) {
+      trailers[trailerName] = reply[kReplyTrailers][trailerName](reply, payload)
+    }
+    res.addTrailers(trailers)
+  }
 
   // avoid ArgumentsAdaptorTrampoline from V8
-  res.end(payload, null, null)
+  res.end(null, null, null)
```

Comment:
```suggestion
  res.end()
```
If there is no more arguments in end, I think is okay to just use `end()`, because no more params type needs to be judged by v8

Reply:
It is the same as line 443.

Reply:
ref: https://github.com/fastify/fastify/pull/813#issuecomment-368696063